File: Dagger2Demo/src/main/java/com/example/dagger2demo/MainActivity.java
package com.example.dagger2demo;

import android.databinding.DataBindingUtil;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.dagger2demo.databinding.ActivityMainBinding;
import com.example.dagger2demo.javaBean.Student;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity
{

	@Inject
	Student student;

	@Override
	protected void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
		/**
		 * 前面两步都配置好了后，Student  、StudentComponent
		 * Dagger2是基于APT编译时注解的框架， 需要先build一次，才会生成DaggerStudentComponent类
		 * 然后就可以使用了，可以查看源码框架生成的代码做了什么
		 */
		DaggerStudentComponent.create().injectStudent(this);
		binding.setStudent(student);
	}
}

File: KotlinBindingDemo/src/main/java/com/example/kotlinbindingdemo/MainActivity.kt
package com.example.kotlinbindingdemo

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.example.kotlinbindingdemo.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.user = User(" Gibson", "Cool", 24)
    }
}
